actions/assistant.py

```diff
@@ -169,13 +169,16 @@ def _install_repo(repo: Dict[str, str], remove_dir: bool = True) -> List[str]:
         if "checkout" in repo:
             assert isinstance(repo["checkout"], str)
             cmds.append(f"git checkout {repo['checkout']}")
-        pip_install = "."
-        if "install_extras" in repo:
-            pip_install += f"[{AssistantCLI._extras(repo['install_extras'])}]"
-        cmds.append(f"pip install --quiet {pip_install}")
-        if "install_file" in repo:
-            assert isinstance(repo["install_file"], str)
-            cmds.append(f"pip install --quiet --upgrade -r {repo['install_file']}")
+        if "install_command" in repo:
+            cmds.append(repo["install_command"])
```

Comment:
interesting idea, could give an example of custom installation? :rabbit:

Reply:
`python setup.py install` / `pip install .` doesn't work for NNI because NNI needs to build typescript files into js.

Unfortunately `python setup.py develop` doesn't work either because `prepare_env.sh` injects an `rm -rf nni` afterwards, which makes development installation crash.

I'm thinking we might be able to build wheel with this script https://github.com/microsoft/nni/blob/master/test/vso_tools/build_wheel.py
But this looks a bit nasty.

Reply:
It looks like tests on every push needs your approval.

I don't know how many attempts I need to make it right...

Reply:
> It looks like tests on every push needs your approval.

this is coming from GH itself, they require to be a contributor or get approval...
kind of bypass is committing a code that is not perfect, and so `pre-commit[bot]` would followup with fixing commit
(`pre-commit[bot]` is already authorised "user")

Reply:
I didn't understand how to use this `pre-commit[bot]`. Any tutorial on this?

Reply:
for example add two blank lines at the end of the file (the correct is just one, so the bot will engage with correction)

Reply:
> Any tutorial on this?

not sure it is in fact how to hack GH's rules :D

Reply:
> My wild guess is that github action has a very limited RAM.

they are quite limited, so we can run it on GPU machine :rabbit: 

> Can I go with development build, and remove the rm -rf at the end of prepare_env.sh script?

Let me see...